### DISCUSSION TITLE
Is there any way to change a component from native code?

### DISCUSSION THREAD

**Discussion by hikmetli** [slint-ui/slint#4620] [Q&A]
I am trying to change a component that created with .slint file or a string. I want to add another component that i have took from another file or string into a component tat already exist. Is there any way to do this in runtime?

**Comment by hunger** [comment]
No, the tree of items is constant. You can have some fairly dynamic sub-trees using models, but that is how far things go right now using the stable APIs we have.

> [!WARNING]
> 
> There is more if you are willing to venture into the unstable and unsupported realm.
>
> There is a `ComponentContainer` used in the live preview, which can load arbitrary Slint UIs at run time and embed that into your UI. But as I said: This needs unstable features to be passed during the build, uses unstable APIs and is Rust-only. It also does not load arbitrary slint UIs just yet: It is limited to load slint files via the interpreter right now (nothing pre-compiled yet!).
>
> I hope to be able to stabilize that container at some point, but it is not a priority for us: We are still find rough spots we need to polish away before we want to see this getting adopted more widely. To become stable, this would obviously also get implementations for other languages.
>
> But you have been warned: API on this *will* change before it becomes stable, probably several times and with no warning. But any help with stabilizing this is of course welcome:-)
>
> **THIS WILL BREAK BEFORE BECOMING STABLE**

**Comment by ogoffart** [comment]
If you use the slint-interpreter crate, you can change the full UI and do some search and replace to change the code at runtime before showing the window. But you can't at the moment do that on an visible UI (without using private unstable API)
